server/server_test.go

```diff
@@ -2121,6 +2121,239 @@ func TestStartPositionTimeDelta(t *testing.T) {
+		stan.DeliverAllAvailable(),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait for message to be received and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+	// Restart a connection
+	sc = NewDefaultConnection(t)
+	defer sc.Close()
+	// Restart the durable with different start position. It should
+	// be ignored and resume from where it left of.
+	// Since connection is closed in the callback last message is not
+	// acked, but that's fine. We still wait for the next message.
+	atomic.StoreUint64(&expected, 2)
+	if _, err := sc.Subscribe("foo", cb,
+		stan.StartAtSequence(uint64(total-5)),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait to receive message and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+	// Restart a connection
+	sc = NewDefaultConnection(t)
+	defer sc.Close()
+	// Try with start sequence above total message
+	atomic.StoreUint64(&expected, 3)
+	if _, err := sc.Subscribe("foo", cb,
+		stan.StartAtSequence(uint64(total+10)),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait to receive message and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+	// Restart a connection
+	sc = NewDefaultConnection(t)
+	defer sc.Close()
+	// Try with start time in future, which should not fail if
+	// start position is ignored for a resumed durable.
+	atomic.StoreUint64(&expected, 4)
+	if _, err := sc.Subscribe("foo", cb,
+		stan.StartAtTime(time.Now().Add(10*time.Second)),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait to receive message and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+}
+
+func TestStartPositionWithDurableQueueSub(t *testing.T) {
+	s := RunServer(clusterName)
+	defer s.Shutdown()
+
+	sc := NewDefaultConnection(t)
+	defer sc.Close()
+	total := 100
+	for i := 0; i < total; i++ {
+		if err := sc.Publish("foo", []byte("msg")); err != nil {
+			t.Fatalf("Unexpected error on publish: %v", err)
+		}
+	}
+	ch := make(chan bool)
+	expected := uint64(1)
+	cb := func(m *stan.Msg) {
+		if m.Sequence == atomic.LoadUint64(&expected) {
+			sc.Close()
+			ch <- true
+		}
+	}
+	// Start a durable queue subscriber at first index
+	if _, err := sc.QueueSubscribe("foo", "group", cb,
+		stan.DeliverAllAvailable(),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait for message to be received and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
```

Comment:
Yes, if you see in the `cb`, I close the connection when the message with expected sequence is received.